api.go

```diff
@@ -590,8 +590,12 @@ func statusContainer(pod *Pod, containerID string) (ContainerStatus, error) {
 				}
 
 				if !running {
-					if err := container.stop(); err != nil {
-						return ContainerStatus{}, err
+					err = container.state.validTransition(container.state.State, StateStopped)
+					if err == nil {
+						err = container.setContainerState(StateStopped)
```

Comment:
I'm not sure I fully understand this - if the shim is killed/dies but the VM is still running, won't this leave stray processes around? Also, if `Container.stop()` isn't being called, the hot-unplug code won't be called (`c.removeDrive()`)?

Reply:
Good point, I'll rework the fix.